### PR TITLE
[MAINT] Pytest fixture to ignore internal deprecation warnings

### LINKED ISSUE #1737: Too many deprecation warnings cluttering the outputs
Running CircleCI, we get a TON of deprecation warnings. This will be a problem in the future when the bill is suddenly due (and we may not forsee the deadline, since there are so many), and right now it makes the already verbose output of CI systems and makes it worse.

We should consider slowly attending to the deprecations as we go about fixing other stuff.

### PR DESCRIPTION
**Pull request by ymzayek**
- With #3936 closes #1737.

Changes proposed in this pull request:

- Add a fixture to suppress known internal deprecation warnings

I'm not sure if this is the best approach, but it avoids having to go through all the tests that raise the warning and adding e.g. a parameter to suppress the warning, which is a bit of a hassle and adds parameters unrelated to the particular unit tests. Since the warnings are known and tested it shouldn't have any negative side effects and could also be a way to track our deprecation in one place.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
I would say that until we find a way to uniformize how we throw / test deprecation warnings (see https://github.com/nilearn/nilearn/issues/3673 and https://github.com/nilearn/nilearn/issues/3625) this seems like a good approach for now.

**Comment by Remi-Gau** [comment]
failures are unrelated, no?

**Comment by bthirion** [comment]
I think they are.

**Comment by Remi-Gau** [comment]
Yup. I relaunched the test on main and everything passed so seems unrelated to some dependency change at least.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: nilearn/conftest.py [hunk 1]
```diff
@@ -112,6 +112,23 @@ def warnings_as_errors():
...
 
 
+@pytest.fixture(autouse=True)
+def suppress_specific_warning():
+    """Ignore internal deprecation warnings."""
+    with warnings.catch_warnings():
+        messages = (
+            "The `darkness` parameter will be deprecated.*|"
+            "`legacy_format` will default to `False`.*|"
+            "In release 0.13, this fetcher will return a dictionary.*|"
```
Comment (ymzayek) [inline nilearn/conftest.py:122]: Thanks for finishing the PR @Remi-Gau !
Just fyi I did not check that this was a thorough list of internal deprecation messages.

Comment (Remi-Gau) [reply nilearn/conftest.py:122]: will quickly check but your work should allow to make it easy to suppress if there are more of those warnings

**Pull request merged**